### PR TITLE
451 use program number service

### LINKED ISSUE #451: Allocate number endpoint
Flesh out `graphql/document/src/mutations/allocate_number.rs` using the number table (using a custom number type)

### PR DESCRIPTION
**Pull request by clemens-msupply**
Closes #451

### CODE REVIEW

File: server/service/src/auth.rs [hunk 1]
```diff
@@ -293,13 +292,6 @@ fn all_permissions() -> HashMap<Resource, PermissionDSL> {
             PermissionDSL::HasPermission(Permission::PatientMutate),
         ]),
     );
-    map.insert(
```
Comment (jmbrunskill) [inline server/service/src/auth.rs:296]: Deliberately removing permissions?

**Comment by jmbrunskill** [comment]
Looks good to me :)
Only question is around if there should be some form of permissions on the endpoint?

File: server/service/src/auth.rs (hunk 1 above)
Comment (jmbrunskill) [reply server/service/src/auth.rs:296]: Ah I see you are using the MutateProgram permission. That seems like a good idea.

Comment (clemens-msupply) [reply server/service/src/auth.rs:296]: Yes I renamed the endpoint and MutateProgram seemed more appropriated.

**Pull request merged**